### PR TITLE
Use @ngrx/signals as the signal store in the frontend

### PR DESCRIPTION
**Pull request by d3473r**

### CODE REVIEW

File: frontend/libs/table/src/lib/table/table-display/table-display.component.ts [hunk 1]
```diff
@@ -16,13 +14,15 @@ import { Game } from '@dig-kick/models';
...
-      .observe(`table/${this.tableName()}/game`)
-      .pipe(map((value) => JSON.parse(value.payload.toString()) as Game)),
-  );
+  readonly gameStore = inject(GameStore);
 
-  constructor(private _mqttService: MqttService) {}
+  game = computed<Game | undefined>(() => {
+    const tableId = this.tableId();
+    if (tableId) {
+      return this.gameStore.entityMap()[tableId];
```
Comment (hiqs-chris) [inline frontend/libs/table/src/lib/table/table-display/table-display.component.ts:24]: Does the library provide an entityMap automatically? It is not defined in game.store

File: frontend/libs/services/src/lib/dk-mqtt-client.service.ts [hunk 2]
```diff
@@ -1,27 +1,90 @@
...
-    }
-    this._mqttService.observe(topic);
-    return this.topics[topic].asObservable();
+  readonly tableStore = inject(TableStore);
+  readonly gameStore = inject(GameStore);
+  readonly playerStore = inject(PlayerStore);
+
+  tablesRaw = toSignal(this._mqttService.observe('tables'));
+
+  scoreRaw = toSignal(this._mqttService.observe(`api/player/all`));
```
Comment (hiqs-chris) [inline frontend/libs/services/src/lib/dk-mqtt-client.service.ts:29]: You just load all players? Shouldn't it only be the players relevant to the current game?

File: frontend/libs/models/src/lib/game.model.ts [hunk 3]
```diff
@@ -1,13 +1,16 @@
...
 }
+
+export const GameSchema = z.object({
+  gameMode: z.nativeEnum(GameMode),
+  teamWhite: TeamSchema.nullish(),
+  teamBlack: TeamSchema.nullish(),
+  pointsToWin: z.number(),
+});
+
+export type Game = z.infer<typeof GameSchema> & { tableId: string };
```
Comment (hiqs-chris) [inline frontend/libs/models/src/lib/game.model.ts:16]: Move the tableId into the schema

**Review by hiqs-chris — CHANGES_REQUESTED** [review]

File: frontend/src/app/app.config.ts [hunk 4]
```diff
@@ -11,7 +11,6 @@ export const appConfig: (environment: Environment) => ApplicationConfig = (
     hostname: environment.broker.hostname,
     port: environment.broker.port,
     path: environment.broker.path,
-    clientId: environment.broker.clientId,
```
[the remark this answers is not in the document]
Comment (d3473r) [reply frontend/src/app/app.config.ts:14]: The clientId is not configurable in the library we use in the frontend

File: frontend/src/app/app.component.ts [hunk 5]
```diff
@@ -18,7 +18,10 @@ import { Theme } from '@dig-kick/models';
   styleUrl: './app.component.scss',
 })
 export class AppComponent {
-  constructor(private themeService: ThemeService) {
+  constructor(
+    private themeService: ThemeService,
+    private dkMqttClientService: DkMqttClientService,
```
[the remark this answers is not in the document]
Comment (d3473r) [reply frontend/src/app/app.component.ts:23]: Added an init method

File: frontend/libs/table/src/lib/table/table-display/table-display.component.ts (hunk 1 above)
Comment (d3473r) [reply frontend/libs/table/src/lib/table/table-display/table-display.component.ts:24]: Yes this is automagically provided: https://ngrx.io/guide/signals/signal-store/entity-management#withentities-feature

File: frontend/libs/services/src/lib/dk-mqtt-client.service.ts (hunk 2 above)
Comment (d3473r) [reply frontend/libs/services/src/lib/dk-mqtt-client.service.ts:32]: This loads all players for the scoreboard

File: frontend/libs/models/src/lib/game.model.ts (hunk 3 above)
Comment (d3473r) [reply frontend/libs/models/src/lib/game.model.ts:16]: This has to be added by the frontend, as the game is no (yet) aware to which table it belongs in the database

**Review by hiqs-chris — APPROVED** [review]

**Pull request merged**